### ISSUE TITLE
In-game graphics only use a quarter of the screen

### ISSUE THREAD

**Issue by a-detiste** [Andrettin/Wyrmsun#42]
I got this bug once, I run in "almost-fullscreen" with radeonsi free drivers

Tried to save the game, close & reload, but this just crash; so restarting from scratch.

![wyrmsun_quarter](https://cloud.githubusercontent.com/assets/7994192/11806668/fb99c510-a314-11e5-9694-8379dfce2cd5.png)

**Comment by Andrettin** [comment]
Thanks for the report! Someone had reported this bug once, but  I unfortunately never managed to reproduce it. If you look at the minimap, only the parts which actually show on screen are within its selection rectangle. Could you upload your save game, so that I could take a look at it? Since it is still crashing, it is quite possible that it contains a clue to solving the bug.

**Comment by a-detiste** [comment]
Here it is

http://users.teledisnet.be/ade15809/game.sav.gz

**Comment by Andrettin** [comment]
Thanks! Unfortunately though I'm getting a "not found" error.

**Comment by a-detiste** [comment]
here's the right one

http://users.skynet.be/bs366950/game.sav.gz

**Comment by Andrettin** [comment]
Thanks! Unfortunately, the issue when loading the game was unrelated, so that the screen looked fine after loading it. In any case, I've fixed the issue which caused the loading crash for the next patch.

**Comment by a-detiste** [comment]
I now see https://github.com/Andrettin/Wyrmsun/issues/18

**Comment by Andrettin** [comment]
Did this happen just after playing the "Tomb of Durahn Eikinskjaldi Quest"? If so, I've just fixed it:
https://github.com/Andrettin/Wyrmsun/commit/3824d2f4fd3d2ce9045df7c05f6da4a46ed2607f

**Comment by a-detiste** [comment]
I don't quite understad: do you mean the game engine status is not fully reset when changing maps ?
I was just playing the map one by one in standard order, except for humans when I missed/skipped one.

**Comment by Andrettin** [comment]
The visible map area used to be reset when loading the UI for a civilization, but this stopped being the case when a lot of code was moved from the civilizations' UI files to the main ui.lua file (to avoid code duplication); the visible map area variables were instead only loaded when starting Wyrmsun or changing resolution. This had the unforeseen consequence of causing the issue with the partially black screen after playing the Tomb of Durahn Eikinskjaldi quest. That quest's map width is smaller than the visible map area width in most resolutions, and this caused the engine to reduce the UI.MapArea.EndX variable accordingly.

I made it so now the UI.MapArea.EndX variable is always reloaded when starting a new scenario.
